daemon-scheduler/pkg/deployment/environment.go

```diff
@@ -161,14 +163,40 @@ func (e environment) FilterEnvironments(ctx context.Context, filterKey string, f
 }
 
 func (e environment) filterEnvironmentsByCluster(ctx context.Context, cluster string) ([]types.Environment, error) {
```

Comment:
what is the use case for the daemon scheduler to retrieve clusters across environments? We should never be deploying to multiple environments at once.